### PR TITLE
publish volume changes for multiple target IQN

### PR DESCRIPTION
**Pull request by wdurairaj**
Updated spec.md for multiple target IQN's in publish volume call

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -365,11 +365,26 @@ PUT `/containers/v1/volumes/{id}/actions/publish`
         ],
         "lun_id": 0,
         "serial_number": "4349bd228896f1236c9ce9006592f26f",
-        "target_name": "iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"
+        "target_names": "iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"
     }
 }
 ```
  * Note that `chap_user` and `chap_password` must also be part of the response if CHAP details were provided as part of the Node definition.
+ * `target_names` json field should be separated by comma, if there are multiple target IQN's needs to be passed to CSI Driver.
```
Comment (shivamerla) [inline spec.md]: @wdurairaj target_names should be returned as an array from CSP and converted to CSV format by CSI driver later.

File: spec.md [hunk 2]
```diff
@@ -365,11 +365,26 @@ PUT `/containers/v1/volumes/{id}/actions/publish`
         ],
         "lun_id": 0,
         "serial_number": "4349bd228896f1236c9ce9006592f26f",
-        "target_name": "iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"
+        "target_names": "iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"
```
Comment (shivamerla) [inline spec.md]: This should be an array with single entry.

**Review by shivamerla — APPROVED** [review]
lgtm..

File: spec.md [hunk 3]
```diff
@@ -365,11 +365,27 @@ PUT `/containers/v1/volumes/{id}/actions/publish`
         ],
         "lun_id": 0,
         "serial_number": "4349bd228896f1236c9ce9006592f26f",
-        "target_name": "iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"
+        "target_names": ["iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"]
     }
 }
 ```
  * Note that `chap_user` and `chap_password` must also be part of the response if CHAP details were provided as part of the Node definition.
+ * `target_names`  should be returned as an array of target IQN's from CSP and converted to CSV format by CSI driver.
```
Comment (shivamerla) [inline spec.md]: nit: This note is not required for target_names.

File: spec.md [hunk 4]
```diff
@@ -355,7 +355,7 @@ PUT `/containers/v1/volumes/{id}/actions/publish`
 }
 ```
 
-#### Response
+#### Response for single target IQN
```
Comment (rgcostea) [inline spec.md]: drop "for single target IQN"

File: spec.md (hunk 3 above)
Comment (rgcostea) [inline spec.md]: nit: IQNs from the CSP

File: spec.md [hunk 5]
```diff
@@ -365,11 +365,27 @@ PUT `/containers/v1/volumes/{id}/actions/publish`
...
         "serial_number": "4349bd228896f1236c9ce9006592f26f",
-        "target_name": "iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"
+        "target_names": ["iqn.2007-11.com.nimblestorage:group-array1-g3b5de80e54af7a6b"]
     }
 }
 ```
  * Note that `chap_user` and `chap_password` must also be part of the response if CHAP details were provided as part of the Node definition.
+ * `target_names`  should be returned as an array of target IQN's from CSP and converted to CSV format by CSI driver.
+ 
+ #### Response for multiple target IQN
```
Comment (rgcostea) [inline spec.md]: nit: IQNs

**Review by rgcostea — CHANGES_REQUESTED** [review]
Please also update the https://github.com/hpe-storage/container-storage-provider/blob/master/spec.md#objects table below.

File: spec.md [hunk 6]
```diff
@@ -655,7 +655,7 @@ This endpoint is used to manage the creation and deletion of snapshot groups tha
 | | serial_number | string | X | | X |
 | | access_protocol | string | X | | X |
 | | lun_id | number | X | | X |
-| | target_name | string | only for iscsi | | X |
+| | target_names | string | only for iscsi | | X |
```
Comment (rgcostea) [inline spec.md]: change string to list\<string\>

**Review by rgcostea — APPROVED** [review]

**Pull request merged**